pkg/types/aws/machinepool.go

```diff
@@ -25,6 +25,11 @@ type MachinePool struct {
 	// +optional
 	EC2RootVolume `json:"rootVolume"`
 
+	// EC2MetadataOptions defines metadata service interaction options for EC2 instances in the machine pool.
+	//
+	// +optional
+	EC2Metadata `json:"metadataService"`
```

Comment:
Don't think we should embed this:
```suggestion
	EC2Metadata EC2Metadata `json:"metadataService"`
```